### PR TITLE
refactor to async/await

### PR DESCRIPTION
**Pull request by fananek**
Implemented async/await for Vapor template.

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]
Thanks!

**Comment by 0xTim** [comment]
As an FYI I'm aware this needs merging, we just probably need to announce this and set up a new branch in case anyone is stuck on macOS 11 or earlier. It's on my TODO list

**Comment by fananek** [comment]
OK Tim, thanks for info. Let me know in case I can help somehow.

File: Sources/App/Controllers/TodoController.swift [hunk 1]
```diff
@@ -11,19 +11,21 @@ struct TodoController: RouteCollection {
         }
     }
 
-    func index(req: Request) throws -> EventLoopFuture<[Todo]> {
-        return Todo.query(on: req.db).all()
+    func index(req: Request) async throws -> [Todo] {
+        return try await Todo.query(on: req.db).all()
```
Comment (0xTim) [inline Sources/App/Controllers/TodoController.swift:15]: Nit
```suggestion
       try await Todo.query(on: req.db).all()
```

File: Sources/App/Migrations/CreateTodo.swift [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import Fluent
 
-struct CreateTodo: Migration {
-    func prepare(on database: Database) -> EventLoopFuture<Void> {
-        return database.schema("todos")
+struct CreateTodo: AsyncMigration {
+    func prepare(on database: Database) async throws {
+        return try await database.schema("todos")
```
Comment (0xTim) [inline Sources/App/Migrations/CreateTodo.swift:5]: Nit
```suggestion
       try await database.schema("todos")
```

File: Sources/App/Migrations/CreateTodo.swift [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+       try await database.schema("todos")
             .id()
             .field("title", .string, .required)
             .create()
     }
 
-    func revert(on database: Database) -> EventLoopFuture<Void> {
-        return database.schema("todos").delete()
+    func revert(on database: Database) async throws {
+        return try await database.schema("todos").delete()
```
Comment (0xTim) [inline Sources/App/Migrations/CreateTodo.swift:12]: Nit
```suggestion
       try await database.schema("todos").delete()
```

File: Sources/App/Migrations/CreateTodo.swift [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+       try await database.schema("todos")
             .id()
             .field("title", .string, .required)
             .create()
     }
 
-    func revert(on database: Database) -> EventLoopFuture<Void> {
-        return database.schema("todos").delete()
+    func revert(on database: Database) async throws {
+       try await database.schema("todos").delete()
```
Comment (0xTim) [inline Sources/App/Migrations/CreateTodo.swift:12]: Spacing
```suggestion
        try await database.schema("todos").delete()
```

File: Sources/App/Migrations/CreateTodo.swift [hunk 5]
```diff
@@ -1,14 +1,14 @@
 import Fluent
 
-struct CreateTodo: Migration {
-    func prepare(on database: Database) -> EventLoopFuture<Void> {
-        return database.schema("todos")
+struct CreateTodo: AsyncMigration {
+    func prepare(on database: Database) async throws {
+       try await database.schema("todos")
```
Comment (0xTim) [inline Sources/App/Migrations/CreateTodo.swift:5]: Spacing
```suggestion
        try await database.schema("todos")
```

File: Sources/App/Controllers/TodoController.swift [hunk 6]
```diff
@@ -11,19 +11,21 @@ struct TodoController: RouteCollection {
...
     }
 
-    func delete(req: Request) throws -> EventLoopFuture<HTTPStatus> {
-        return Todo.find(req.parameters.get("todoID"), on: req.db)
-            .unwrap(or: Abort(.notFound))
-            .flatMap { $0.delete(on: req.db) }
-            .transform(to: .ok)
+    func delete(req: Request) async throws -> HTTPStatus {
+        guard let todo = try await Todo.find(req.parameters.get("todoID"), on: req.db) else {
+                throw Abort(.notFound)
```
Comment (0xTim) [inline Sources/App/Controllers/TodoController.swift:26]: Spacing
```suggestion
            throw Abort(.notFound)
```

File: Sources/App/Controllers/TodoController.swift [hunk 7]
```diff
@@ -11,19 +11,21 @@ struct TodoController: RouteCollection {
...
+            }
```
Comment (0xTim) [inline Sources/App/Controllers/TodoController.swift:27]:
```suggestion
        }
```

File: Sources/App/Controllers/TodoController.swift [hunk 8]
```diff
@@ -11,19 +11,21 @@ struct TodoController: RouteCollection {
         }
     }
 
-    func index(req: Request) throws -> EventLoopFuture<[Todo]> {
-        return Todo.query(on: req.db).all()
+    func index(req: Request) async throws -> [Todo] {
+       try await Todo.query(on: req.db).all()
```
Comment (0xTim) [inline Sources/App/Controllers/TodoController.swift:15]:
```suggestion
        try await Todo.query(on: req.db).all()
```

**Pull request merged**